app/recipe/views.py

```diff
@@ -0,0 +1,18 @@
+from rest_framework.viewsets import ModelViewSet
+
+from recipe.models import Recipe
+from recipe.serializers import RecipeSerializer
+
+
+class RecipeViewSet(ModelViewSet):
+    serializer_class = RecipeSerializer
+    queryset = Recipe.objects.all()
+
+    def get_queryset(self):
+        name = self.request.query_params.get('name')
+
+        queryset = self.queryset
+        if name:
+            queryset = queryset.filter(name__startswith=name)
```

Comment:
Nice usage of the framework. Keep in mind that since you are searching just for a word you are not interested to be "case sensitive" in the search, so maybe it will be more useful to use the [istartswith](https://docs.djangoproject.com/en/dev/ref/models/querysets/#istartswith) filter that make the exact same thing but it's case insensitive.

Reply:
Yup, it should be better case-insensitive.